src/color.rs

```diff
@@ -70,24 +72,83 @@ impl Colors {
             Theme::NoColor => None,
             Theme::Default => Some(Self::get_light_theme_colour_map()),
         };
+        let lscolors = LsColors::from_env().unwrap_or_default();
 
-        Self { colors }
+        Self { colors, lscolors }
     }
 
     pub fn colorize<'a>(&self, input: String, elem: &Elem) -> ColoredString<'a> {
         self.style(elem).paint(input)
     }
 
+    pub fn colorize_using_path<'a>(&self, input: String, path: &str) -> ColoredString<'a> {
+        self.style_from_path(path).paint(input)
```

Comment:
You should check if the `Self::lscolors` variable is set. If not, you can fallback on `Self::colorize`